stellar/deploy-contract/commands.js

```diff
@@ -7,97 +7,108 @@ const { mainProcessor, upgrade, upload, deploy } = require('./processors');
 
 require('../cli-utils');
 
-const CONTRACT_DEPLOY_OPTIONS = {
-    AxelarGateway: () => [
-        new Option('--nonce <nonce>', 'optional nonce for the signer set'),
-        new Option('--domain-separator <domainSeparator>', 'domain separator (keccak256 hash or "offline")').default('offline'),
-        new Option('--previous-signers-retention <previousSignersRetention>', 'previous signer retention').default(15).argParser(Number),
-        new Option('--minimum-rotation-delay <miniumRotationDelay>', 'minimum rotation delay').default(0).argParser(Number),
-    ],
-    AxelarExample: () => [
-        new Option('--use-dummy-its-address', 'use dummy its address for AxelarExample contract to test a GMP call').default(false),
-    ],
-    InterchainTokenService: () => [
-        new Option('--interchain-token-version <interchainTokenVersion>', 'version for InterchainToken contract').makeOptionMandatory(true),
-        new Option('--token-manager-version <tokenManagerVersion>', 'version for TokenManager contract').makeOptionMandatory(true),
-    ],
-    InterchainToken: () => [
-        new Option('--name <name>', 'token name (e.g., "Test Token")').makeOptionMandatory(true).argParser(String),
-        new Option('--symbol <symbol>', 'token symbol (e.g., "TEST")').makeOptionMandatory(true).argParser(String),
-        new Option('--decimals <decimals>', 'token decimals (e.g., 7)').makeOptionMandatory(true).argParser(parseInt),
-    ],
+const CONTRACT_CONFIG = {
+    AxelarGateway: {
+        deployOptions: [
+            ['--nonce <nonce>', 'optional nonce for the signer set'],
+            ['--domain-separator <domainSeparator>', 'domain separator (keccak256 hash or "offline")', { default: 'offline' }],
+            ['--previous-signers-retention <previousSignersRetention>', 'previous signer retention', { default: 15, parser: Number }],
+            ['--minimum-rotation-delay <miniumRotationDelay>', 'minimum rotation delay', { default: 0, parser: Number }],
```

Comment:
this is adding an unnecessary layer of abstraction. now there is a bespoke interface of option config that needs to converted to the right `Option` object. Why not just keep specifying `new Option` as before. It's not really making things more concise, and if we need to use an extra feature of `Option`, now we need additional wrapper logic.